packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -953,6 +953,44 @@ class Auth {
         return await this.loginGuestUser()
     }
 
+    /**
+     * Handle updating customer password and re-log in after the access token is invalidated.
+     *
+     */
+    async updateCustomerPassword(body: {
+        customer: ShopperCustomersTypes.Customer
+        password: string
+        currentPassword: string
+    }) {
+        const {
+            customer: {customerId, login},
+            password,
+            currentPassword
+        } = body
+
+        // login and customerId are optional fields on the Customer type
+        // here we had to guard it to avoid ts error
+        if (!login || !customerId) {
+            throw new Error('Customer is missing required fields.')
+        }
+
+        const res = await this.shopperCustomersClient.updateCustomerPassword({
+            headers: {
+                authorization: `Bearer ${this.get('access_token')}`
+            },
+            parameters: {customerId},
+            body: {
+                password: password,
+                currentPassword: currentPassword
+            }
+        })
+        await this.loginRegisteredUserB2C({
+            username: login,
+            password
+        })
```

Comment:
should we make this configurable to let users to choose whether they want to log the user out, or silently re-login the registered user?

Reply:
A toggle has been added to handle the silent re-login. If the user does not opt for the re-login, they are taken to the login screen.

If the silent re-login errors out for some reason, the user will remain on the account page. Then when the app makes a request with the now invalid access token, it will be logged in as a guest. In other words, it will behave the same as another session in other browser.